tukaan/_props.py

```diff
@@ -19,182 +26,145 @@ def config(widget: TkWidget, **kwargs) -> None:
     Tcl.call(None, widget, "configure", *Tcl.to_tcl_args(**kwargs))
 
 
-def cget(widget: TkWidget, return_type: Any, option: str) -> Any:
+def cget(widget: TkWidget, return_type: type[T], option: str) -> T:
     return Tcl.call(return_type, widget, "cget", option)
 
 
-def get_image_pos(self) -> ImagePosition:
-    return cget(self, ImagePosition, "-compound")
-
-
-def set_image_pos(self, value: ImagePosition) -> None:
-    config(self, compound=value)
-
-
-image_pos = property(get_image_pos, set_image_pos)
-
-
-def get_text_align(self) -> Justify:
-    return cget(self, Justify, "-justify")
-
-
-def set_text_align(self, value: Justify) -> None:
-    config(self, justify=value)
-
-
-text_align = property(get_text_align, set_text_align)
-
-
-def get_fg_color(self) -> Color:
-    return cget(self, Color, "-foreground")
-
-
-def set_fg_color(self, value: Color | str) -> None:
-    config(self, foreground=value)
-
-
-fg_color = property(get_fg_color, set_fg_color)
-
-
-def get_bg_color(self) -> Color:
-    return cget(self, Color, "-foreground")
-
-
-def set_bg_color(self, value: Color | str) -> None:
-    config(self, foreground=value)
-
+class RWProperty(Protocol[T_co, T_contra]):
+    def __get__(self, instance: TkWidget, owner: object = None) -> T_co:
+        ...
 
-bg_color = property(get_bg_color, set_bg_color)
+    def __set__(self, instance: TkWidget, value: T_contra) -> None:
+        ...
 
 
-def get_text(self) -> str:
-    return cget(self, str, "-text")
+class WidgetDesc(RWProperty[T, T_contra]):
+    def __init__(self, command: str, type: type[T]):
+        self._command = command
+        self._type = type
 
+    def __get__(self, instance: TkWidget, owner: object = None) -> T:
+        if owner is None:
+            return NotImplemented
+        return cget(instance, self._type, f"-{self._command}")
 
-def set_text(self, value: str) -> None:
-    config(self, text=value)
+    def __set__(self, instance: TkWidget, value: T_contra):
+        config(instance, **{self._command: value})
 
 
-text = property(get_text, set_text)
+class BoolDesc(WidgetDesc[bool, bool]):
+    def __init__(self, command: str):
+        super().__init__(command, bool)
 
 
-def get_width(self) -> int:
-    return cget(self, int, "-width")
+class FloatDesc(WidgetDesc[float, float]):
+    def __init__(self, command: str):
+        super().__init__(command, float)
 
 
-def set_width(self, value: int) -> None:
-    config(self, width=value)
+class Compound(WidgetDesc[ImagePosition, ImagePosition]):
```

Comment:
Yess, I see. It would cause a name conflict with the `ImagePosition` enum.
However then having `Compound` and `ImagePosition` in the same scope can be confusing. We should try to indicate in such cases, that it's a property descriptor.

Reply:
ImagePosition is already taken by the enum. Alternative names like ImagePos etc. look bad. Compound sure looks unituitive. How about ImageAnchor or something?

Reply:
Imo, having both `ImageAnchor` and `ImagePosition` is really confusing. I'd go with `ImagePositionProp`